### PR TITLE
Add support for initdb scripts

### PR DESCRIPTION
**Pull request by dhinus**
This pull request adds the possibility to run SQL files and sh files in /docker-entrypoint-initdb.d -- this is similar to how the [official Docker image](https://hub.docker.com/_/postgres/) works and is useful to preload the database with seed data.

Other changes:

* use default TCP port (9092 instead of 1521)
* add documentation to README.md

## Test Plan

I built the image locally with these changes, then built a separate image extending this one:

Dockerfile:
```
FROM buildo/h2database

ADD *.sql /docker-entrypoint-initdb.d/
ADD *.sh /docker-entrypoint-initdb.d/
ADD baz /docker-entrypoint-initdb.d/
```

Output of `docker build -t h2test . && docker run --rm -it -p 8082:8082 h2test`:

```
Sending build context to Docker daemon    194kB
Step 1/4 : FROM buildo/h2database
 ---> 008f944a7bd2
Step 2/4 : ADD *.sql /docker-entrypoint-initdb.d/
 ---> ea650c76f707
Step 3/4 : ADD *.sh /docker-entrypoint-initdb.d/
 ---> 485638f57e98
Step 4/4 : ADD baz /docker-entrypoint-initdb.d/
 ---> 9b4947383699
Successfully built 9b4947383699
Successfully tagged h2test:latest

/usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/baz

/usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-initdb.d/init.sh

/usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-initdb.d/od-persistent.sql
using url jdbc:h2:/h2-data/od-persistent

/usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-initdb.d/od-volatile.sql
using url jdbc:h2:/h2-data/od-volatile

TCP server running at tcp://172.17.0.3:9092 (others can connect)
Web Console server running at http://172.17.0.3:8082 (others can connect)
```

I then verified that connecting to the web console I could see all the databases and tables created by the init scripts:

screen shot 2018-01-24 at 12 59 37

screen shot 2018-01-24 at 12 58 14

screen shot 2018-01-24 at 12 58 49

screen shot 2018-01-24 at 12 59 24

### CODE REVIEW

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -1,5 +1,32 @@
...
+  filename=$(basename "$1")
+  db="${filename%.*}"
+  url="jdbc:h2:$H2DATA/$db"
+  echo "using url $url"
+  $RUNSCRIPT -script "$1" -url "$url"
+}
+
+mkdir -p "$H2DATA"
+
+if [ ! -f "$H2DATA/.initdb.completed" ]; then
```
Comment (danielegallingani) [inline docker-entrypoint.sh]: ❤️

**Comment by danielegallingani** [comment]
💯

**Comment by markgetafe** [comment]
Is this pull request going to be merged?

**Comment by gabro** [comment]
@markgetafe hey, what happened here is that we dropped H2 for most of our projects so we didn't push this forward since we don't need this anymore at work.

That said I think we can merge it and publish a new version if that's helpful

**Pull request merged**